confidential-data-hub/storage/src/volume_type/alibaba_cloud_oss/oss.rs

```diff
@@ -0,0 +1,142 @@
+    #[serde(rename = "targetPath")]
+    pub target_path: String,
+    pub url: String,
+    #[serde(rename = "volumeId")]
+    pub volume_id: String,
+}
+
+async fn unseal_secret(secret: Vec<u8>) -> Result<Vec<u8>> {
+    // TODO: verify the jws signature using the key specified by `kid`
+    // in header. Here we directly get the JWS payload
+    let payload = secret.split(|c| *c == b'.').nth(1).ok_or_else(|| {
+        Error::SecureMountFailed("illegal input sealed secret (not a JWS)".into())
+    })?;
+
+    let secret_json = STANDARD.decode(payload).map_err(|e| {
+        Error::SecureMountFailed(format!(
+            "illegal input sealed secret (JWS body is not standard base64 encoded): {e}"
+        ))
+    })?;
+    let secret: Secret = serde_json::from_slice(&secret_json).map_err(|e| {
+        Error::SecureMountFailed(format!(
+            "illegal input sealed secret format (json deseralization failed): {e}"
+        ))
+    })?;
+
+    let res = secret
+        .unseal()
+        .await
+        .map_err(|e| Error::SecureMountFailed(format!("unseal failed: {e}")))?;
+    Ok(res)
+}
+
+async fn get_plain(secret: String) -> Result<String> {
+    if secret.starts_with("sealed.") {
+        let tmp = secret
+            .strip_prefix("sealed.")
+            .ok_or(anyhow!("strip_prefix \"sealed.\" failed"))
+            .map_err(|e| {
+                Error::SecureMountFailed(format!("strip_prefix \"sealed.\" failed: {e}"))
+            })?;
+        let unsealed = unseal_secret(tmp.into())
+            .await
+            .map_err(|e| Error::SecureMountFailed(format!("unseal secret failed: {e}")))?;
+
+        return String::from_utf8(unsealed)
+            .map_err(|e| Error::SecureMountFailed(format!("convert to String failed: {e}")));
+    }
+    Err(Error::SecureMountFailed(format!(
+        "sealed secret format error!"
+    )))
+}
+
+impl Oss {
+    pub(crate) async fn mount(&self, source: String, mount_point: String) -> Result<String> {
+        let plain_ak_id = get_plain(self.ak_id.clone())
+            .await
+            .map_err(|e| Error::SecureMountFailed(format!("get_plain failed: {e}")))?;
+        let plain_ak_secret = get_plain(self.ak_secret.clone())
+            .await
+            .map_err(|e| Error::SecureMountFailed(format!("get_plain failed: {e}")))?;
+        if self.encrypted == "gocryptfs" {
+            let plain_passwd = get_plain(self.enc_passwd.clone())
+                .await
+                .map_err(|e| Error::SecureMountFailed(format!("get_plain failed: {e}")))?;
+            Command::new(MOUNT_SCRIPT)
+                .arg("-i")
+                .arg(plain_ak_id)
+                .arg("-s")
+                .arg(plain_ak_secret)
+                .arg("-b")
+                .arg(self.bucket.clone())
+                .arg("-p")
+                .arg(self.path.clone())
+                .arg("-o")
+                .arg(self.other_opts.clone())
+                .arg("-d")
+                .arg(source)
```

Comment:
`-d` is the `mountpoint` according to the script. It's a bit confusing to me that it is set to the `source` here.

Reply:
mount.sh is removed.